app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/MoveNodeFragment.kt

```diff
@@ -0,0 +1,207 @@
+        viewLifecycleScope.launch {
+            val action = MoveNodeAction(pos)
+            elementEditsController.add(editType, node, ElementPointGeometry(node.position), "survey", action)
+            listener?.onMovedNode(editType, pos)
+        }
+    }
+
+    override fun onClickMapAt(position: LatLon, clickAreaSizeInMeters: Double) = false
+
+    @UiThread override fun onMapMoved(position: LatLon) {
+        if (checkIsDistanceOkAndUpdateText(position))
+            okButtonContainer.popIn()
+        else
+            okButtonContainer.popOut()
+    }
+
+    private fun checkIsDistanceOkAndUpdateText(position: LatLon): Boolean {
+        val moveDistance = position.distanceTo(node.position)
+        return when {
+            moveDistance < MIN_MOVE_DISTANCE -> {
+                binding.titleLabel.setText(R.string.node_moved_not_far_enough)
+                false
+            }
+            moveDistance > MAX_MOVE_DISTANCE -> {
+                binding.titleLabel.setText(R.string.node_moved_too_far)
+                false
+            }
+            else -> {
+                binding.titleLabel.text = resources.getString(R.string.node_moved,displayUnit.format(moveDistance.toFloat()))
+                true
+            }
+        }
+    }
+
+    @UiThread override fun onClickClose(onConfirmed: () -> Unit) {
+        if (!hasChanges) {
+            onConfirmed()
+        } else {
+            activity?.let {
+                AlertDialog.Builder(it)
+                    .setMessage(R.string.confirmation_discard_title)
+                    .setPositiveButton(R.string.confirmation_discard_positive) { _, _ ->
+                        onConfirmed()
+                    }
+                    .setNegativeButton(R.string.short_no_answer_on_button, null)
+                    .show()
+            }
+        }
+    }
+
+    companion object {
+        private const val ARG_NODE = "node"
+        private const val ARG_QUEST_TYPE = "quest_type"
+
+        fun create(elementEditType: ElementEditType, node: Node): MoveNodeFragment {
+            val f = MoveNodeFragment()
+            f.arguments = bundleOf(
+                ARG_NODE to Json.encodeToString(node),
+                ARG_QUEST_TYPE to elementEditType.name
+            )
+            return f
+        }
+    }
+}
+
+// Require a minimum distance for two reasons:
+// 1. The map is not perfectly precise, especially displayed road widths may be off by a few meters,
+//     so it may be hard to tell whether something really is misplaced (e.g. bench along a path)
+//     without good aerial imagery.
+// 2. The value added by moving nodes by such small distance, even if correct, is rather low.
+private const val MIN_MOVE_DISTANCE = 2.0
+// Move node functionality is meant for fixing slightly misplaced elements. If something moved far
+// away, it is reasonable to assume there are more substantial changes required, also to nearby elements.
+private const val MAX_MOVE_DISTANCE = 30.0
```

Comment:
Another reason: the related elements are only shown in a certain radius. The move distance may not be larger than that radius.